src/backend/catalog/index.c

```diff
@@ -950,8 +950,12 @@ index_create(Relation heapRelation,
 		}
 		else
 		{
-			indexRelationId =
-				GetNewRelFileNode(tableSpaceId, pg_class, relpersistence);
+			/* Index OIDs must be kept in normal OID range. */
+			if (is_enr && GetNewPermanentRelFileNode_hook)
+				indexRelationId = GetNewPermanentRelFileNode_hook(tableSpaceId, pg_class, relpersistence);
```

Comment:
this approach has following issue
In future if there is any change in GetNewRelFileNode, it will not come into GetNewPermannent... function and subtle bugs can creep in. I think it would be better to modify function with one additional argument. We need to change at 3-4 other places only. One benefit of this is also that in future if someone adds a new call to GetNewRelFileNode, they need to pass a new argument and it will be detected at merge time.
Now for function GetNewRelFileNode, we have another option of keeping it unchanged with just a hook call at beginning into extension with modified logic. This will make engine side changes to bare minimum. However, it also has disadvantage that a change on engine side might get missed on corresponding hook on extension side.